[#45084] Add menu items to navigate to activity page showing project attributes

cbliard opened the pull request:
https://community.openproject.org/work_packages/45084

cbliard committed:
Add acceptance tests

cbliard committed:
Add 'Project activity' menu item in projects list

ulferts commented:
The `dev` branch had failing specs from whence you branched off. Please rebase/merge to fix this.

mrmir committed:
Add link to project activity page when clicking on menu item

cbliard committed:
Add failing acceptance test for Project details activity

cbliard committed:
Add 'Project details activity' menu entry

Added to the Project details widget on project overview page.

Displaying the menu entry conditionally depending on the activity module
being activated is done through a public permission. Capabilities
endpoint has been extended to include actions from public permissions.

cbliard committed:
refactor: Make test faster

5.3 secs => 2.1 secs

cbliard committed:
refactor: Make spec example names more descriptive

cbliard committed:
[#45084] Check permission instead of module activation

cbliard committed:
refactor: Improve readability with explicit parameter names

b12f reviewed (approved):
LGTM on the frontend side of things, though there's some linting errors

dombesz commented on spec/features/projects/projects_index_spec.rb:
```diff
@@ -1007,4 +1008,55 @@ def expect_projects_in_order(*projects)
+
+      it 'redirects to project activity page with only project attributes visible' do
+        login_as(simple_member)
+        visit projects_path
```
I think these 2 specs can be merged into one, saving the time to reload the page. WDYT?

cbliard replied:
Good point. Done.

dombesz commented on spec/models/capabilities/scopes/default_spec.rb:
```diff
@@ -101,117 +90,181 @@
-      let(:user_status) { Principal.statuses[:locked] }
+      context 'with the user being locked' do
+        before do
+          user.locked!
```
The `locked!` method seems to be deprecated.
https://github.com/opf/openproject/blob/07927d1ce62c7125e3898a7ca293767cc93a83f2/app/models/user.rb#L316

cbliard replied:
`lock!` is deprecated, and `locked!` should be used intead.

`deprecated_alias` signature is `deprecated_alias(old_method, new_method)`.

I suggest changing it to `deprecated_alias(dont_use:, use:)`, so that using it would look like
```ruby
  deprecated_alias dont_use: :lock!, use: :locked!
```

WDYT?

dombesz commented on spec/models/capabilities/scopes/default_spec.rb:
```diff
@@ -101,117 +90,181 @@
       end
-    end
-
-    context 'with a global member with an action permission and the user being locked' do
```
Is this covered somewhere else?

cbliard replied:
Yes. It may not appear clearly: I nested some tests in other existing context to better differentiate the normal case and the special cases. Now it's on line 155.

```
Capabilities::Scopes::Default
  .default
    with a global member with a global permission
      includes the actions of the global permission for the scoped to user
      with the user being locked                                               <==== *here*
        is empty for the scoped to user
```

dombesz reviewed (approved):
LGTM! 👍 I left a few questions, but feel free to merge once those are cleared.

cbliard committed:
Make test faster by doing more in one go

cbliard commented:
Ignoring eslint errors as they were there before.
Merging as there are no other errors.
